psa-crypto/src/types/algorithm.rs

```diff
@@ -685,3 +720,42 @@ impl From<AsymmetricSignature> for psa_crypto_sys::psa_algorithm_t {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use crate::types::algorithm::{Algorithm, AsymmetricSignature, Hash, SignHash};
+    use core::convert::TryInto;
+
+    #[test]
+    fn conversion() {
+        assert_eq!(
+            Hash::Sha256,
+            psa_crypto_sys::PSA_ALG_SHA_256.try_into().unwrap()
+        );
+        assert_eq!(psa_crypto_sys::PSA_ALG_SHA_256, Hash::Sha256.into());
+        assert_eq!(
+            SignHash::Any,
+            psa_crypto_sys::PSA_ALG_ANY_HASH.try_into().unwrap()
+        );
+        assert_eq!(
+            SignHash::Specific(Hash::Sha256),
+            psa_crypto_sys::PSA_ALG_SHA_256.try_into().unwrap()
+        );
+        assert_eq!(
+            Algorithm::AsymmetricSignature(AsymmetricSignature::Ecdsa {
+                hash_alg: SignHash::Specific(Hash::Sha3_512),
+            }),
+            psa_crypto_sys::PSA_ALG_ECDSA(psa_crypto_sys::PSA_ALG_SHA3_512)
+                .try_into()
+                .unwrap()
+        );
+        assert_eq!(
+            psa_crypto_sys::PSA_ALG_ECDSA(psa_crypto_sys::PSA_ALG_SHA3_512),
+            Algorithm::AsymmetricSignature(AsymmetricSignature::Ecdsa {
+                hash_alg: SignHash::Specific(Hash::Sha3_512),
+            })
+            .try_into()
+            .unwrap()
+        );
```

Comment:
Could you add some checks that fail (i.e. with `unwrap_err()`)?

Reply:
Yes, will add!